### ISSUE TITLE
Can't compile

### ISSUE THREAD

**Issue by MotikaCraft-TheJoshua** [ds58/Panilla#172]
Hello there
i am (desperately) trying to compile the plugin despite having it purchased, i love to have both premium and compiled versions of the softwares i purchase (and not purchase, because i usually do it for any other cool plugin i come across)

It fails when it comes to paper-v1_20_6 (and i assume it will also at paper-v1_20_6)

Any chance you could fix this? I don't know what i didn't try anymore....

Thank you in advance

**Comment by professorparabellum** [comment]
I am LITERALLY in the exact same boat, Im so pissed off that they pulled builds from spigot and maintain it but dont leave some sort of CI server open or something. I paid specifically so i wouldnt have to deal with this

**Comment by MotikaCraft-TheJoshua** [comment]
> I am LITERALLY in the exact same boat, Im so pissed off that they pulled builds from spigot and maintain it but dont leave some sort of CI server open or something. I paid specifically so i wouldnt have to deal with this

All we can do is wait for @ds58 to reply and hopefully help us
I really tried everything
I tried multiple gradle versions, wrappers, settings, i also tried locally importing paper.jar but i didn't search out more so i could find how to actually find remapped PAPER jars (i know that i can obtain spigot ones via BuildTools which i always do)
I tried multiple java versions
Changing options in build.gradle for java versions
Changing configuration for repos used (i needed to add 2-3 more than currently are in build.gradle)
etc...

**Comment by professorparabellum** [comment]
I almost turned racist trying to compile this: [Panilla.zip](https://github.com/user-attachments/files/18143937/Panilla.zip)
Don't download jars from random people.

**Comment by MotikaCraft-TheJoshua** [comment]
> I almost turned racist trying to compile this: [Panilla.zip](https://github.com/user-attachments/files/18143937/Panilla.zip) Don't download jars from random people.

Can you tell me how did you compile?

**Comment by professorparabellum** [comment]
I'm gonna be completely honest i have no idea. I think what I did was clone the paper 1.20.6 and latest branch, compile it, and then ran the paperclip jar. I added the paperclip jar itself, the jar inside the versions/ folder, and recursively added every single jar file inside of the libraries/ folder. I did this for both versions

**Comment by MotikaCraft-TheJoshua** [comment]
Where you added:
- paperclip jar
- jar generated when starting the paperclip inside of `versions/`
- every single jar file inside of the `libraries/` after running paperclip jar

So where did you put all this and also did you modify gradle build files for Panilla and how specifically?

**Comment by Kawi16** [comment]
If you want, there’s a (compiled) version in the fork I made with the latest commits. Later, if I have time, I will upload the updated Gradle files and I will make a pull request.

**Comment by MotikaCraft-TheJoshua** [comment]
> If you want, there’s a (compiled) version in the fork I made with the latest commits. Later, if I have time, I will upload the updated Gradle files and I will make a pull request.

Please do man - upload updated gradle files when you can - and enable actions in there
Also i would like to donate if you do this so leave me your paypal email here

**Comment by Kawi16** [comment]
Done! 
Btw, I don't want anything, I do this for passion and as a hobby.

**Comment by MotikaCraft-TheJoshua** [comment]
> Done! Btw, I don't want anything, I do this for passion and as a hobby.

I will remember your name. :) <3

**Comment by MotikaCraft-TheJoshua** [comment]
I tried applying everything as you did,
it seems like i have very similar errors as before:
https://mclo.gs/qaspMlU

**Comment by Kawi16** [comment]
> I tried applying everything as you did,
> 
> it seems like i have very similar errors as before:
> 
> https://mclo.gs/qaspMlU

Have you buildtools installed?

**Comment by MotikaCraft-TheJoshua** [comment]
> > I tried applying everything as you did,
> > it seems like i have very similar errors as before:
> > https://mclo.gs/qaspMlU
> 
> Have you buildtools installed?

Can you tell me how exactly i need to install it?
Should i just put BuildTools.jar inside of the repo and run it to generate 1.20.6 & 1.21.1 server jars?

**Comment by Kawi16** [comment]
> Can you tell me how exactly i need to install it?
>
> Should i just put BuildTools.jar inside of the repo and run it to generate 1.20.6 & 1.21.1 server jars?

You need to have all the required build tools because you need the Spigot libraries. To do this faster, create a shell script.

**Comment by W1CK3DWORKS** [comment]
> > Can you tell me how exactly i need to install it?
> > Should i just put BuildTools.jar inside of the repo and run it to generate 1.20.6 & 1.21.1 server jars?
> 
> You need to have all the required build tools because you need the Spigot libraries. To do this faster, create a shell script.

I already fixed this by adding additional maven entry
It compiles correctly for 1.8.8 and 1.12.2-1.20.4 SPIGOT
Only the paper 1.20.6 and 1.21.1 is problem for me, you can see on the logs link provided

**Comment by Kawi16** [comment]
> I already fixed this by adding additional maven entry
> 
> It compiles correctly for 1.8.8 and 1.12.2-1.20.4 SPIGOT
> 
> Only the paper 1.20.6 and 1.21.1 is problem for me, you can see on the logs link provided
> 
> 
With the Paperweight plugin, you should be able to compile the plugin (see my fork: https://github.com/Kawi16/Panilla)
If you’re unable to do so, message me on Discord (@kawi16) and let me know here once you’ve sent me a friend request.

**Comment by MotikaCraft-TheJoshua** [comment]
> > I already fixed this by adding additional maven entry
> > It compiles correctly for 1.8.8 and 1.12.2-1.20.4 SPIGOT
> > Only the paper 1.20.6 and 1.21.1 is problem for me, you can see on the logs link provided
> 
> With the Paperweight plugin, you should be able to compile the plugin (see my fork: https://github.com/Kawi16/Panilla) If you’re unable to do so, message me on Discord (@Kawi16) and let me know here once you’ve sent me a friend request.

Added you, @W1CK3DWORKS and I are in team

**Comment by Kawi16** [comment]
Issue resolved!

**Comment by MotikaCraft-TheJoshua** [comment]
Solved by building 1.20.6 & 1.21.1 spigots using BuildTools inside of the Panilla cloned repo, then building Panilla regularly using gradle:
```bash
java -jar BuildTools.jar --rev 1.20.6 --remapped
java -jar BuildTools.jar --rev 1.21.1 --remapped
```
Thanks to @Kawi16
